src/test/java/seedu/address/logic/commands/EditAssignmentCommandTest.java

```diff
@@ -0,0 +1,116 @@
+                originalAssignment.getName(), studentToEdit.getName().fullName, VALID_SCORE),
+                result.getFeedbackToUser());
+    }
+
+    @Test
+    void execute_invalidStudentIndex_throwsCommandException() {
+        // Initialize a command with an out-of-bounds student index
+        EditAssignmentCommand command = new EditAssignmentCommand(Index.fromZeroBased(1),
+                INDEX_FIRST_ASSIGNMENT, VALID_SCORE);
+
+        // Expect a CommandException due to invalid student index
+        CommandException thrown = assertThrows(CommandException.class, () -> command.execute(model));
+        assertEquals(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX, thrown.getMessage());
+    }
+
+    @Test
+    void execute_invalidAssignmentIndex_throwsCommandException() {
+        // Retrieve the student but leave their assignment list empty
+        Student studentToEdit = model.getFilteredStudentList().get(INDEX_FIRST_STUDENT.getZeroBased());
+        studentToEdit.setAssignmentList(new ArrayList<>()); // clear assignments for this test
+
+        // Initialize a command with an out-of-bounds assignment index
+        EditAssignmentCommand command = new EditAssignmentCommand(INDEX_FIRST_STUDENT,
+                Index.fromZeroBased(1), VALID_SCORE);
+
+        // Expect a CommandException due to invalid assignment index
+        CommandException thrown = assertThrows(CommandException.class, () -> command.execute(model));
+        assertEquals(Messages.MESSAGE_INVALID_ASSIGNMENT_INDEX, thrown.getMessage());
+    }
+
+    @Test
+    void equals() {
+        final EditAssignmentCommand standardCommand = new EditAssignmentCommand(INDEX_FIRST_STUDENT,
+                INDEX_FIRST_ASSIGNMENT, VALID_SCORE);
+
+        // Same values should return true
+        EditAssignmentCommand commandWithSameValues = new EditAssignmentCommand(INDEX_FIRST_STUDENT,
+                INDEX_FIRST_ASSIGNMENT, VALID_SCORE);
+        assertTrue(standardCommand.equals(commandWithSameValues));
+
+        // Same object should return true
+        assertTrue(standardCommand.equals(standardCommand));
+
+        // Null should return false
+        assertTrue(!standardCommand.equals(null));
+
+        // Different types should return false
+        assertTrue(!standardCommand.equals(new Object()));
+
+        // Different student index should return false
+        assertTrue(!standardCommand.equals(new EditAssignmentCommand(Index.fromZeroBased(2),
+                INDEX_FIRST_ASSIGNMENT, VALID_SCORE)));
+
+        // Different assignment index should return false
+        assertTrue(!standardCommand.equals(new EditAssignmentCommand(INDEX_FIRST_STUDENT,
+                Index.fromZeroBased(2), VALID_SCORE)));
+
+        // Different score should return false
+        assertTrue(!standardCommand.equals(new EditAssignmentCommand(INDEX_FIRST_STUDENT, INDEX_FIRST_ASSIGNMENT,
+                90)));
+    }
+}
```

Comment:
Maybe could add a test for the toString method just for increasing test coverage